### PR TITLE
Dropout for DAG

### PR DESCRIPTION
**Pull request by miaecle**
This PR adds:

- Fix some doc strings and arguments in `DAGLayer`

- Initial support for dropout in DAG model

Due to the special structure of DAG, the dropout structure added here is a stepwise process that happens on each step(propagate through one directed edge). Another approach is only doing dropout on  the outputs, or equivalently the last step of propagation. Comments and suggestions on the potential structure choice here are very welcomed!

### CODE REVIEW

File: deepchem/models/tensorgraph/graph_layers.py [hunk 1]
```diff
@@ -698,11 +699,13 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
       self.out_tensor = out_tensor
     return out_tensor
 
-  def DAGgraph_step(self, batch_inputs, W_list, b_list):
+  def DAGgraph_step(self, batch_inputs, W_list, b_list, **kwargs):
     outputs = batch_inputs
     for idw, W in enumerate(W_list):
       outputs = tf.nn.xw_plus_b(outputs, W, b_list[idw])
       outputs = self.activation(outputs)
+      if 'training' in kwargs and kwargs['training'] == 1.0 and not self.dropout is None:
```
Comment (peastman) [inline deepchem/models/tensorgraph/graph_layers.py]: `training` will often be a tensor, not a number, so this comparison will fail.  Just multiply the dropout rate by it and don't worry about the value (like in the Dropout layer).

Comment (miaecle) [reply deepchem/models/tensorgraph/graph_layers.py]: Good point, just made the change!

File: deepchem/models/tensorgraph/models/graph_models.py [hunk 2]
```diff
@@ -381,6 +384,12 @@ def __init__(self,
               Number of features for atom in the graph
             n_outputs: int, optional
               Number of features for each molecule
+            layer_sizes: list of int, optional
+              Structure of hidden layer(s) in DAG propagation step
```
[the remark this answers is not in the document]
Comment (miaecle) [reply deepchem/models/tensorgraph/models/graph_models.py]: Sure, added a more detailed explanation here.

**Comment by peastman** [comment]
Looks good.
